pkg/ext-proc/handlers/request.go

```diff
@@ -116,6 +119,7 @@ func (s *Server) HandleRequestBody(reqCtx *RequestContext, req *extProcPb.Proces
 			},
 		},
 	}
+	metrics.MonitorRequest(llmReq.Model, llmReq.ResolvedTargetModel, len(v.RequestBody.Body), time.Since(requestReceivedTimestamp))
```

Comment:
I haven't read through the PR but sharing this early feedback.

request start time: to be precise, this should be the first time when the `Process`  function is called, perhaps here https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/1ad93e4b79d6752ff11fa92b8a10fbaf13b4b47e/pkg/ext-proc/handlers/server.go#L55. You can use the `RequestContext` to store states for a request.

request end time: This needs to be the time after we finish processing the response body, potentially here: https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/1ad93e4b79d6752ff11fa92b8a10fbaf13b4b47e/pkg/ext-proc/handlers/server.go#L87

Reply:
Updated to record the metrics in the processing loop.